File: server/schema/camera.py
from uuid import UUID
from pydantic import BaseModel
from enum import Enum

class CamType(Enum):
    RTSP = 'RTSP'
    IP = 'IP'
class Status(Enum):
    ACTIVE = 'ACTIVE'
    INACTIVE = 'INACTIVE'
class Camera(BaseModel):
    ip_address: str
    cam_type: CamType
    username: str | None
    password: str | None
    port: int
    channel: int | None
    subtype: int | None
    latitude: float
    longitude: float
    azimuth: int
    status: Status
    thana: str

class Search(BaseModel):
    lat: float
    lon: float
    radius: float

File: client/src/types/camera.ts
export enum Status{
    ACTIVE = 'ACTIVE',
    INACTIVE = 'INACTIVE'
}

export interface Camera {
    id: string;
    username: string;
    ipAddress: string;
    password: string;
    port: number;
    channel: number;
    subType: number;
    latitude: number;
    longitude: number;
    azimuth: number;
    status: Status;
    thana: string;
}